File: LordOfTheGames/app/src/main/java/com/example/lordofthegames/Database/CardItemRepository.java
package com.example.progetto.Database;

import android.app.Application;

import androidx.lifecycle.LiveData;

import com.example.progetto.CardItem;

import java.util.List;

public class CardItemRepository {
    private final CardItemDAO cardItemDAO;

    private final LiveData<List<CardItem>> cardItemList;

    public CardItemRepository(Application application){
        CardItemDatabase db = CardItemDatabase.getDatabase(application);
        cardItemDAO = db.cardItemDAO();

        cardItemList = cardItemDAO.getCardItems();
    }


    // Room executes all queries on a separate thread.
    // Observed LiveData will notify the observer when the data has changed.
    public LiveData<List<CardItem>> getCardItemList() {
        return cardItemList;
    }

    // You must call this on a non-UI thread or your app will throw an exception. Room ensures
    // that you're not doing any long running operations on the main thread, blocking the UI.
    public void addCardItem(CardItem cardItem) {
        CardItemDatabase.executor.execute(new Runnable() {
            @Override
            public void run() {
                cardItemDAO.addCardItem(cardItem);
            }
        });
    }
}

File: LordOfTheGames/app/src/main/java/com/example/lordofthegames/Database/CardItemRepository.kt
package com.example.lordofthegames.Database

import android.app.Application
import androidx.lifecycle.LiveData
import com.example.lordofthegames.recyclerView.CardItem

class                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        nCardItemRepository(application: Application?) {
    private lateinit var cardItemDAO: CardItemDAO
    val cardItemList: LiveData<List<CardItem?>?>?

    init {
        val db: CardItemDatabase? = application?.let { CardItemDatabase.getDatabase(it) }
        if (db != null) {
            cardItemDAO = db.cardItemDAO()!!
        }
        cardItemList = cardItemDAO.getCardItems()
    }

    fun addCardItem(cardItem: CardItem?) {
        CardItemDatabase.executor.execute(Runnable { cardItemDAO.addCardItem(cardItem) })
    }
}